source/math/Quaternion.ooc

```diff
@@ -91,7 +107,79 @@ Quaternion: cover {
 		else
 			roll = asin((upYawPitch scalarProduct(upRotated) * upYawPitch z - upRotated z) / rightYawPitch z)
 
-		FloatRotation3D new(pitch, -yaw, roll)
+		FloatRotation3D new (pitch, -yaw, roll)
+	}
+	norm: Float { get { (this real squared() + (this imaginary norm) squared()) sqrt() } }
```

Comment:
A shorter syntax for `function: ReturnType { get { code } }` is `function ::= code` (see lines 26-29, for example)